Documentation/user-guides/configuring-cluster-monitoring.md

```diff
@@ -1,137 +1,629 @@
+For more in-depth documentation on how to configure audit please refer to [OpenShift Docs setting audit log levels for prometheus adapter](https://docs.openshift.com/container-platform/latest/monitoring/configuring-the-monitoring-stack.html#setting-audit-log-levels-for-the-prometheus-adapter_configuring-the-monitoring-stack).
+
+Example
+
+```yaml
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: cluster-monitoring-config
+  namespace: openshift-monitoring
+data:
+  config.yaml: |
+    k8sPrometheusAdapter:
+      audit:
+        profile: request
+      nodeSelector:
+        kubernetes.io/os: linux
+      tolerations:
+        - operator: Exists
+```
+
+### kubeStateMetrics
+
+Holds configuration for all things kube-state-metrics agent.
+
+```yaml
 kubeStateMetrics:
-  baseImage: custom-registry.com/kube-state-metrics
-  addonResizerBaseImage: custom-registry.com/addon-resizer
+  # Define which Nodes the Pods are scheduled on.
+  nodeSelector: v1.NodeSelector
+  # Defines the Pods tolerations.
+  tolerations: v1.Toleration
 ```
 
-> Note: The container images coming from repositories of a custom registry are expected to mirror the canonical repositories on [quay.io][quay].
+Example
 
-## Reference
+```yaml
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: cluster-monitoring-config
+  namespace: openshift-monitoring
+data:
+  config.yaml: |
+    kubeStateMetrics:
+      nodeSelector:
+        kubernetes.io/os: linux
+      tolerations:
+        - operator: Exists
+```
+
+### prometheusK8s
 
-The following configuration options are available for Cluster Monitoring.
+Holds configuration related to the Prometheus component that will be managed by prometheus-operator.
 
-### Config
+```yaml
+prometheusK8s:
+  # Holds configuration about how the Prometheus component should communicate with Alertmanager,
+  # from configuring the URL scheme, timeout to authentication method and others.
+  additionalAlertmanagerConfigs: <struct>
+  # Enforces body size limit of Prometheus scrapes, if a scrape is bigger than
+  # the limit it will fail.
+  # 3 kinds of values are accepted:
+  #  1. empty value: no limit
+  #  2. a value in Prometheus size format, e.g. "64MB"
+  #  3. string "automatic", which means the limit will be automatically calculated based on
+  #     cluster capacity.
+  enforcedBodySizeLimit: <string>
+  # The labels to add to any time series or alerts when communicating with external systems
+  # (federation, remote storage, Alertmanager).
+  externalLabels: <struct>
+  # Log level for Prometheus to be configured with.
+  # Possible values are: error, warn, info, debug.
+  logLevel: <string>
+  # Define which Nodes the Pods are scheduled on.
+  nodeSelector: v1.NodeSelector
+  # QueryLogFile specifies the file to which PromQL queries are logged. Suports both just a filename
+  # in which case they will be saved to an emptyDir volume at /var/log/prometheus, if a full path is
+  # given an emptyDir volume will be mounted at that location.
```

Comment:
I think that the rules are more complicated than that so we need to be more exhaustive.

Reply:
Humm I'm not quite sure what I might be missing. I've added regarding relative paths and linux std streams. Let me know if I'm missing something